src/main/scala/org/constellation/ConstellationNode.scala

```diff
@@ -366,6 +380,11 @@ class ConstellationNode(
       .unsafeRunSync()
   }
 
+  def setTokenForGenesisNode = {
+    logger.info("Creating token for genesis node")
+    dao.sessionTokenService.createAndSetNewOwnToken.unsafeRunAsync(_ => ())
```

Comment:
should it be async? if so, it creates the token in background so if you're using it in chain computation, you cannot rely on that, that just after that line you have token prepared. also, async without context shift can block the thread

Reply:
I must say I mimicked the code around to some extent. It doesn't need to exist immediately. It needs to exist when first peer will try to join to genesis.